extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTenantConfigResolver.java

```diff
@@ -20,9 +20,9 @@
     @Inject
     Instance<TenantConfigResolver> tenantConfigResolver;
 
-    private Map<String, TenantConfigContext> tenantsConfig;
-    private TenantConfigContext defaultTenant;
-    private Function<OidcTenantConfig, TenantConfigContext> tenantConfigContextFactory;
+    private volatile Map<String, TenantConfigContext> tenantsConfig;
+    private volatile TenantConfigContext defaultTenant;
+    private volatile Function<OidcTenantConfig, TenantConfigContext> tenantConfigContextFactory;
```

Comment:
So I assume this is the part that actually fixes #6608?

Reply:
I hope so. Without this the HTTP thread could read incorrect config, so was likely missing the principalClaim. This is the only explanation I can see for the type of failure we are seeing.

Reply:
I see, thanks for the explanation. Let's see what CI thinks :)